packages/compass-e2e-tests/helpers/commands/scroll-to-virtual-item.ts

```diff
@@ -0,0 +1,53 @@
+import type { CompassBrowser } from '../compass-browser';
+
+export async function scrollToVirtualItem(
+  browser: CompassBrowser,
+  itemsSelector: string,
+  targetSelector: string
+): Promise<boolean> {
+  let found = false;
+
+  let lastId: string;
+
+  // Start by scrolling down, then scroll back up. This is necessary in case we
+  // started off somewhere in the middle and the item we're looking for is above
+  // it.
+  for (const direction of [1, -1]) {
+    await browser.waitUntil(async () => {
+      const targetElement = await browser.$(targetSelector);
+      if (await targetElement.isExisting()) {
+        await targetElement.waitForDisplayed();
+        await targetElement.scrollIntoView();
+        // the item is now visible, so stop scrolling
+        found = true;
+        return true;
+      }
+
+      const elements = await browser.$$(itemsSelector);
+      const edgeElement = elements[direction === 1 ? elements.length - 1 : 0];
+      await edgeElement.waitForDisplayed();
+      await edgeElement.scrollIntoView();
+
+      const thisId = await edgeElement.getAttribute('data-id');
```

Comment:
Can't we save a reference to the element directly instead of relying on `data-id`?

Reply:
how? every time you look up the element it is a different reference.

Reply:
> every time you look up the element it is a different reference

I see, is it not like a real dom tree with the same selector returning always the same node?